### ISSUE TITLE
A very peculiar error associated with 'accent' marking preceding a vowel

### ISSUE THREAD

**Issue by Andhrabharati** [sanskrit-lexicon/COLOGNE#407]
Just by chance landed at an MW entry, which led me in identifying this error!!

The occurrences of  accent marks followed by vowels (in the text data) have **few types of  display errors** like (a) missing letters or (b) taking  huge time etc.
--if not involving textual corrections [whether typing or printing errors]-- in the following works:
**CAE, CCS, GRA, INM, MCI, MW, PW, PWG, PWKVN, SCH and STC.**

In the INM, MCI etc. the '/' character is not an accent mark, but still the timing issue is noticed.

Incidentally, the first search result display for these is taking "HELL lot of a time" (sorry for the bad wording!!), few tens of seconds to few minutes in some cases [apparently going into an 'eternal' loop].

The KRM, BOP etc. which have the ^X^ as the superscript notation seem to have no such error.
`------------------------------------`
This case-insensitive regex may be used to get the list-- `[^<][/\\\^][fxaiueo]` (in SLP1)

[I recall notifying one similar error [involving the '|' (slp) character], which was corrected by @funderburkjim (when a reminder directly addressing to him is posted) after few months' of my posting of the issue in the initial days of my looking at MW at CDSL.]

**Comment by Andhrabharati** [comment]
I am not sure if this error occurs in the other works as well.

**Comment by funderburkjim** [comment]
@Andhrabharati   Please provide a specific example so I can reproduce the error.

**Comment by Andhrabharati** [comment]
One quick example for the missing letter after the 'accent marker'--

![image](https://user-images.githubusercontent.com/75209130/226708853-dad23aec-5bb5-4f38-9cd0-a09fe899f9ee.png)

```
<L>9201<pc>1-0689<k1>AreaGa<k2>Are/aGa
{#Are/aGa#}¦ ({#Are + aGa#}) <lex>adj.</lex> {%wovon Uebel fern ist%}: {#A\rea^GA a\sme Ba\drA sO^Srava\sAni^ santu#} 
<ls>ṚV. 6,1,12.</ls> {#sva\stim#} 
<ls n="ṚV. 6,">56,6.</ls>
<LEND>
```

![image](https://user-images.githubusercontent.com/75209130/226709203-61a374f6-f877-4242-a749-281bb295ec44.png)

[As already indicated earlier, I have seen that many a times the regex (as given above) results needed textual corrections in the files.]

**Comment by Andhrabharati** [comment]
![image](https://user-images.githubusercontent.com/75209130/226710843-e2bb9e2c-2866-4ad9-bc76-0a216cf039e1.png)

![image](https://user-images.githubusercontent.com/75209130/226710981-a51760cf-5e4b-49ee-893c-2846e1a4e46a.png)

![image](https://user-images.githubusercontent.com/75209130/226711142-598c5c55-9dd9-477f-93fb-214d47386bd6.png)

**Comment by funderburkjim** [comment]
This problem is peculiar to the PW dictionaries.
Here is a little test:
```
slp1 = Are/agra, deva = आरे॑अग्र, deva1 = आरे꣫ग्र
slp1 = ita/Uti, deva = इत॑ऊति, deva1 = इत꣫ूति
slp1 = go/agra, deva = गो॑अग्र, deva1 = गो꣫ग्र
slp1 = go/fjIka, deva = गो॑ऋजीक, deva1 = गो꣫ृजीक
```

* The 'deva1' field is the transcoding of the slp1 field using the 'slp1_deva1.xml' transcoding file.
  * the deva1 transcoding is used for dictionaries PWG, PW, and PWKVN.
* The 'deva' field  uses the 'slp1_deva.xml' transcoding
   * the deva transcoding is used in all the other dictionaries

Based on the small test:
* the deva transcoding handles the test inputs correctly
* the deva1 transcoding handles the test inputs incorrectly.

My memory is that the PWx transcoding was developed in order to display the accents (notably udAtta) in the manner of Boetlingk.

@Andhrabharati  Can you find a link to the repository and issue where this PWG devanagari was discussed?

The task now is to correct slp1_deva1.xml.

**Comment by Andhrabharati** [comment]
is this (https://github.com/sanskrit-lexicon/PWG/issues/5#issuecomment-900759930)  the one you wanted, @funderburkjim ?

**Comment by Andhrabharati** [comment]
BTW, my above post is not just about the missing letters, it is about the typo/print errors, and the timing issue as well, which I had seen in many works as listed above.

See for example,
![image](https://user-images.githubusercontent.com/75209130/227005550-467e9d69-5cb8-44c0-a308-7571fc06fafd.png)

[the errors are either in the metaline or the following headerline HW entry; and sometimes in the body matter as well.]

**Comment by funderburkjim** [comment]
## changes to slp1_deva1.xml
After this change, the little test looks correct for deva1
```
slp1 = Are/agra, deva = आरे॑अग्र, deva1 = आरे꣫अग्र
slp1 = ita/Uti, deva = इत॑ऊति, deva1 = इत꣫ऊति
slp1 = go/agra, deva = गो॑अग्र, deva1 = गो꣫अग्र
slp1 = go/fjIka, deva = गो॑ऋजीक, deva1 = गो꣫ऋजीक
```

The pwg, pw, and pwkvn displays are changed, including in simple-search

<img width="585" alt="image" src="https://user-images.githubusercontent.com/6393033/227049422-0d5284ec-1b62-458e-8f02-b2f7f08b0536.png">

I will consider this part of the n-fold problem of this issue finished.

**Comment by funderburkjim** [comment]
## (b) taking huge time etc.

@Andhrabharati  please provide a specific example (or a couple of examples) so I can reproduce the problem.

**Comment by funderburkjim** [comment]
## count of the regex
[count.txt](https://github.com/sanskrit-lexicon/COLOGNE/blob/master/issues/407/count.txt) counts the  instances matching the regex `[^<][/\\^][fxaiueoFXAIUEO]` in the 37 dictionaries of csl-orig.

There are 0 instances in 19 of the dictionaries.
A next step would be to look for errors (and non-error patterns) in the others .

**Comment by Andhrabharati** [comment]
> ## (b) taking huge time etc.
> 
> @Andhrabharati  please provide a specific example (or a couple of examples) so I can reproduce the problem.
> 
> 

@funderburkjim 
Now I could not get the error, which I had noticed for the words with the above regex.

I did check it for two days when I posted the issue; and the search for any other word(s) was instantaneous at CDSL and every other site was normal. So, I am sure it was not a network/connection issue at that time.

Probably, we may ignore the timing issue for now.
[If it ever occurs again, it can then be looked into.]
